File: fm-domain/src/main/java/io/zeitwert/fm/ddd/model/SearchResult.java

package io.zeitwert.fm.ddd.model;

import java.math.BigDecimal;

import io.dddrive.ddd.model.enums.CodeAggregateType;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class SearchResult implements Comparable<SearchResult> {

	private Integer tenantId;
	private CodeAggregateType aggregateType;
	private Integer id;
	private String caption;
	private BigDecimal rank;

	@Override
	public int compareTo(SearchResult other) {
		return this.rank.compareTo(other.getRank());
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/ddd/model/SearchResult.kt
package io.zeitwert.fm.ddd.model

import dddrive.ddd.core.model.enums.CodeAggregateType
import java.math.BigDecimal

data class SearchResult(
	val tenantId: Int,
	val aggregateType: CodeAggregateType,
	val id: Int,
	val caption: String,
	val rank: BigDecimal,
) : Comparable<SearchResult> {

	override fun compareTo(other: SearchResult): Int = this.rank.compareTo(other.rank)

}
